### PR TITLE
1067 improve abm tests

### LINKED ISSUE #1143: TestModelTestingCriteria.testAddingAndUpdatingAndRunningTestingSchemes has GMOCK warnings
### Bug description

The test throws warnings, it is possibly miss-configured or otherwise outdated. 

### Version

Any

### To reproduce

Build and run tests, then view TestModelTestingCriteria.testAddingAndUpdatingAndRunningTestingSchemes.

### Relevant log output

```zsh
[----------] 1 test from TestModelTestingCriteria
[ RUN      ] TestModelTestingCriteria.testAddingAndUpdatingAndRunningTestingSchemes

GMOCK WARNING:
/home/runner/work/memilio/memilio/cpp/tests/test_abm_model.cpp:465: Actions ran out in EXPECT_CALL(mock_uniform_dist.get_mock(), invoke)...
Called 3 times, but only 2 WillOnce()s are specified - returning default value.
Stack trace:

GMOCK WARNING:
/home/runner/work/memilio/memilio/cpp/tests/test_abm_model.cpp:465: Actions ran out in EXPECT_CALL(mock_uniform_dist.get_mock(), invoke)...
Called 4 times, but only 2 WillOnce()s are specified - returning default value.
Stack trace:

GMOCK WARNING:
/home/runner/work/memilio/memilio/cpp/tests/test_abm_model.cpp:465: Actions ran out in EXPECT_CALL(mock_uniform_dist.get_mock(), invoke)...
Called 5 times, but only 2 WillOnce()s are specified - returning default value.
Stack trace:
[       OK ] TestModelTestingCriteria.testAddingAndUpdatingAndRunningTestingSchemes (0 ms)
[----------] 1 test from TestModelTestingCriteria (0 ms total)
```

### Add any relevant information, e.g. used compiler, screenshots.

_No response_

### Checklist

- [X] Attached labels, especially loc:: or model:: labels.
- [X] Linked to project

### PR DESCRIPTION
**Pull request by khoanguyen-dev**
# Changes and Information

Please **briefly list the changes** (main added features, changed items, or corrected bugs) made:

- Split up TestWorld.evolveMigration.
- Moved around some tests for better fit
- Add fuzzing or more test cases to TestPerson and TestLocation.
- Add all checks for different VirusVariants in check_constraints() of parameters.h.
- Fix GMOCK warnings of one TestingStrategy test

If need be, add additional information and what the reviewer should look out for in particular:

-

Benchmark results:

- Current branch:
Run on (8 X 24.1214 MHz CPU s)
CPU Caches:
  L1 Data 64 KiB (x8)
  L1 Instruction 128 KiB (x8)
  L2 Unified 4096 KiB (x2)
Load Average: 2.14, 2.46, 2.84
---------------------------------------------------------------------------
Benchmark                                 Time             CPU   Iterations
---------------------------------------------------------------------------
abm_benchmark/abm_benchmark_50k        1300 ms         1294 ms            1
abm_benchmark/abm_benchmark_100k       2599 ms         2586 ms            1
abm_benchmark/abm_benchmark_200k       5220 ms         5187 ms            1

- Main branch:
Run on (8 X 24.0948 MHz CPU s)
CPU Caches:
  L1 Data 64 KiB (x8)
  L1 Instruction 128 KiB (x8)
  L2 Unified 4096 KiB (x2)
Load Average: 3.26, 3.54, 3.26
---------------------------------------------------------------------------
Benchmark                                 Time             CPU   Iterations
---------------------------------------------------------------------------
abm_benchmark/abm_benchmark_50k        1311 ms         1306 ms            1
abm_benchmark/abm_benchmark_100k       2611 ms         2607 ms            1
abm_benchmark/abm_benchmark_200k       5242 ms         5225 ms            1

## Merge Request - Guideline Checklist

Please check our [git workflow](https://github.com/SciCompMod/memilio/wiki/git-workflow). Use the **draft** feature if the Pull Request is not yet ready to review.

### Checks by code author

- [X] Every addressed issue is linked (use the "Closes #ISSUE" keyword below)
- [X] New code adheres to [coding guidelines](https://github.com/SciCompMod/memilio/wiki/Coding-guidelines)
- [X] No large data files have been added (files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [X] Tests are added for new functionality and a local test run was successful (with and without OpenMP)
- [X] Appropriate **documentation** for new functionality has been added (Doxygen in the code and Markdown files if necessary)
- [X] Proper attention to licenses, especially no new third-party software with conflicting license has been added
- [X] (For ABM development) Checked [benchmark results](https://github.com/SciCompMod/memilio/wiki/Agent-Based-Model-Development) and ran and posted a local test above from before and after development to ensure performance is monitored.

### Checks by code reviewer(s)

- [x] Corresponding issue(s) is/are linked and addressed
- [x] Code is clean of development artifacts (no deactivated or commented code lines, no debugging printouts, etc.)
- [x] Appropriate **unit tests** have been added, CI passes, code coverage and performance is acceptable (did not decrease)
- [x] No large data files added in the whole history of commits(files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [ ] On merge, add 2-5 lines with the changes (main added features, changed items, or corrected bugs) to the merge-commit-message. This can be taken from the **briefly-list-the-changes** above (best case) or the separate commit messages (worst case).

Closes #1143

### CODE REVIEW

**Review by reneSchm — COMMENTED** [review]
The changes to parameters look fine, only the loop got changed.
The additional comments in and on the test are really nice, especially on the more complicated ones. Though I am unsure whether the doxygen format should be used on a TEST makro over a regular comment.
This definitely improves the tests, but I will keep the linked issue open for now.

File: cpp/tests/test_abm_person.cpp [hunk 1]
```diff
@@ -77,18 +85,39 @@ TEST(TestPerson, change_location)
...
     EXPECT_EQ(person.get_assigned_location(mio::abm::LocationType::Work), mio::abm::LocationId(4));
+
+    // Fuzzing: assign random valid LocationId values and verify correctness.
+    for (int i = 0; i < 100; ++i) {
+        auto random_id = mio::UniformIntDistribution<int>::get_instance()(rng, 0, 1000);
+        person.set_assigned_location(mio::abm::LocationType::Work, mio::abm::LocationId(random_id));
+        EXPECT_EQ(person.get_assigned_location(mio::abm::LocationType::Work), mio::abm::LocationId(random_id));
+    }
+
+    // Boundary test cases: test with boundary LocationIds.
```
[the remark this answers is not in the document]
Comment (DavidKerkmann) [reply cpp/tests/test_abm_person.cpp:110]: I changed the number above to max(), I think the tests for the boundary cases are fine and described accordingly.

**Review by reneSchm — APPROVED** [review]
